python/pyarrow/tests/test_types.py

```diff
@@ -857,6 +857,31 @@ def test_decimal_overflow():
             pa.decimal256(i, 0)
 
 
+def test_timedelta_overflow():
+    # microsecond resolution, overflow
+    d = datetime.timedelta(days=-106751992, seconds=71945, microseconds=224192)
+    with pytest.raises(pa.ArrowInvalid):
+        pa.scalar(d)
+
+    # microsecond resolution, overflow
+    d = datetime.timedelta(days=106751992, seconds=14454, microseconds=775807)
```

Comment:
This one will overflow based on the days alone. We can also add a case that will only overflow when adding the microseconds. I think `datetime.timedelta(days=106751991, seconds=14454, microseconds=775808)` should to the trick (`datetime.timedelta(days=106751991, seconds=14454, microseconds=775807)` is the limit that works, so adding one more to the microseconds)